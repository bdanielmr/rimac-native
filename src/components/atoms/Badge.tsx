import { LinearGradient } from "expo-linear-gradient";
import React from "react";
import styled from "styled-components/native";

const BadgeContainer = styled.View`
  align-self: flex-start;
  border-radius: 4px;
  overflow: hidden;
  margin-bottom: 8px;
`;

const BadgeGradient = styled(LinearGradient)`
  padding: 6px 10px;
  border-radius: 4px;
`;

const BadgeText = styled.Text`
  color: #03050F;
  font-weight: 700;
  font-size: 12px;
`;

interface BadgeProps {
  children: React.ReactNode;
  testID?: string;
}

export const Badge: React.FC<BadgeProps> = ({ children, testID }) => (
  <BadgeContainer testID={testID}>
    <BadgeGradient
      colors={["#00F4E2", "#00FF7F"]}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 0 }}
    >
      <BadgeText>{children}</BadgeText>
    </BadgeGradient>
  </BadgeContainer>
);