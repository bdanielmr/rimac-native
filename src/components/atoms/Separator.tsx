import styled from "styled-components/native";

export const Separator = styled.View`
  height: 1px;
  background-color: #e5e7eb;
  width: 100%;
  margin-bottom: 24px;
`;