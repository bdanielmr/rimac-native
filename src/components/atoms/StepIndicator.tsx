import React from "react";
import styled from "styled-components/native";

const StepContainer = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 20px 16px;
`;

const StepCircle = styled.View<{ active: boolean; completed: boolean }>`
  width: 24px;
  height: 24px;
  border-radius: 12px;
  background-color: ${(props: any) =>
    props.active ? "#4F4FFF" : props.completed ? "transparent" : "transparent"};
  align-items: center;
  justify-content: center;
  border-width: 1px;
  border-color: ${(props: any) =>
    props.active ? "#4F4FFF" : props.completed ? "#7981B2" : "#7981B2"};
`;

const StepNumber = styled.Text<{ active: boolean; completed: boolean }>`
  color: ${(props: any) => 
    props.active ? "white" : props.completed ? "#7981B2" : "#7981B2"};
  font-weight: 700;
  font-size: 12px;
`;

const StepLabel = styled.Text<{ active: boolean; completed: boolean }>`
  font-size: 16px;
  font-weight: ${(props: any) => (props.active ? "700" : "400")};
  color: ${(props: any) => 
    props.active ? "#141938" : props.completed ? "#7981B2" : "#7981B2"};
`;

const SeparatorIndicator = styled.Text<{ completed: boolean }>`
  color: ${(props: any) => (props.completed ? "#7981B2" : "#4F4FFF")};
  margin: 0 16px;
  font-size: 20px;
`;

const StepWrapper = styled.View`
  flex-direction: row;
  align-items: center;
  gap: 12px;
`;

interface Step {
  number: number;
  label: string;
}

interface StepIndicatorProps {
  steps: Step[];
  currentStep: number;
}

export const StepIndicator: React.FC<StepIndicatorProps> = ({ steps, currentStep }) => (
  <StepContainer>
    {steps.map((step, index) => {
      const isActive = currentStep === step.number;
      const isCompleted = currentStep > step.number;
      
      return (
        <React.Fragment key={step.number}>
          <StepWrapper>
            <StepCircle active={isActive} completed={isCompleted}>
              <StepNumber active={isActive} completed={isCompleted}>
                {step.number}
              </StepNumber>
            </StepCircle>
            <StepLabel active={isActive} completed={isCompleted}>
              {step.label}
            </StepLabel>
          </StepWrapper>
          {index < steps.length - 1 && (
            <SeparatorIndicator completed={isCompleted}>
              - - - -
            </SeparatorIndicator>
          )}
        </React.Fragment>
      );
    })}
  </StepContainer>
);