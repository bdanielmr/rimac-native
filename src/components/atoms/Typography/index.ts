import styled from "styled-components/native";

export const Title = styled.Text`
  font-size: 32px;
  font-weight: 700;
  color: #03050f;
  margin-bottom: 8px;
  line-height: 40px;
  max-width: 352px;
`;

export const MobileTitle = styled.Text`
  font-size: 28px;
  font-weight: 700;
  color: #03050f;
  margin-bottom: 8px;
  line-height: 36px;
`;

export const Subtitle = styled.Text`
  color: #03050f;
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 15px;
  max-width: 352px;
  font-weight: 600;
`;