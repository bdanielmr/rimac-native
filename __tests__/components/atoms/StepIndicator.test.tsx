import { render } from '@testing-library/react-native';
import React from 'react';
import { StepIndicator } from '../../../src/components/atoms/StepIndicator';

describe('StepIndicator Component', () => {
  const mockSteps = [
    { number: 1, label: 'Planes y coberturas' },
    { number: 2, label: 'Resumen' },
  ];

  it('renders all steps correctly', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    expect(getByText('1')).toBeTruthy();
    expect(getByText('Planes y coberturas')).toBeTruthy();
    expect(getByText('2')).toBeTruthy();
    expect(getByText('Resumen')).toBeTruthy();
  });

  it('renders separator between steps', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    expect(getByText('- - - -')).toBeTruthy();
  });

  it('does not render separator after last step', () => {
    const { getAllByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    const separators = getAllByText('- - - -');
    expect(separators).toHaveLength(1); // Solo 1 separador para 2 pasos
  });

  it('highlights current step correctly', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    const currentStepNumber = getByText('1');
    const currentStepLabel = getByText('Planes y coberturas');
    
    expect(currentStepNumber).toBeTruthy();
    expect(currentStepLabel).toBeTruthy();
  });

  it('shows completed state for previous steps', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={2} />
    );

    expect(getByText('1')).toBeTruthy();
    expect(getByText('2')).toBeTruthy();
    expect(getByText('Planes y coberturas')).toBeTruthy();
    expect(getByText('Resumen')).toBeTruthy();
  });

  it('renders with single step', () => {
    const singleStep = [{ number: 1, label: 'Único paso' }];
    const { getByText, queryByText } = render(
      <StepIndicator steps={singleStep} currentStep={1} />
    );

    expect(getByText('1')).toBeTruthy();
    expect(getByText('Único paso')).toBeTruthy();
    expect(queryByText('- - - -')).toBeNull(); // No separator for single step
  });

  it('renders with multiple steps', () => {
    const multipleSteps = [
      { number: 1, label: 'Paso 1' },
      { number: 2, label: 'Paso 2' },
      { number: 3, label: 'Paso 3' },
    ];
    const { getByText, getAllByText } = render(
      <StepIndicator steps={multipleSteps} currentStep={2} />
    );

    expect(getByText('1')).toBeTruthy();
    expect(getByText('2')).toBeTruthy();
    expect(getByText('3')).toBeTruthy();
    expect(getAllByText('- - - -')).toHaveLength(2); // 2 separators for 3 steps
  });

  it('handles first step as current', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    expect(getByText('1')).toBeTruthy();
    expect(getByText('Planes y coberturas')).toBeTruthy();
  });

  it('handles last step as current', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={2} />
    );

    expect(getByText('2')).toBeTruthy();
    expect(getByText('Resumen')).toBeTruthy();
  });

  it('renders with long step labels', () => {
    const longLabelSteps = [
      { number: 1, label: 'Este es un paso con un texto muy largo' },
      { number: 2, label: 'Otro paso con descripción extensa' },
    ];
    const { getByText } = render(
      <StepIndicator steps={longLabelSteps} currentStep={1} />
    );

    expect(getByText('Este es un paso con un texto muy largo')).toBeTruthy();
    expect(getByText('Otro paso con descripción extensa')).toBeTruthy();
  });

  it('renders all step numbers correctly', () => {
    const { getByText } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );

    mockSteps.forEach(step => {
      expect(getByText(step.number.toString())).toBeTruthy();
    });
  });

  it('matches snapshot for first step', () => {
    const { toJSON } = render(
      <StepIndicator steps={mockSteps} currentStep={1} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for last step', () => {
    const { toJSON } = render(
      <StepIndicator steps={mockSteps} currentStep={2} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

});