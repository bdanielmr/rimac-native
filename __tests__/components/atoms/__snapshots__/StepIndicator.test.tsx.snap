// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator Component Snapshot tests matches snapshot for desktop first step 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingBottom": 20,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <View
      active={true}
      completed={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4F4FFF",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderColor": "#4F4FFF",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Text
        active={true}
        completed={false}
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        1
      </Text>
    </View>
    <Text
      active={true}
      completed={false}
      style={
        {
          "color": "#141938",
          "fontSize": 16,
          "fontWeight": "700",
        }
      }
    >
      Planes y coberturas
    </Text>
  </View>
  <Text
    completed={false}
    style={
      {
        "color": "#4F4FFF",
        "fontSize": 20,
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      }
    }
  >
    - - - -
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <View
      active={false}
      completed={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderColor": "#7981B2",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Text
        active={false}
        completed={false}
        style={
          {
            "color": "#7981B2",
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        2
      </Text>
    </View>
    <Text
      active={false}
      completed={false}
      style={
        {
          "color": "#7981B2",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Resumen
    </Text>
  </View>
</View>
`;

exports[`StepIndicator Component Snapshot tests matches snapshot for desktop last step 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingBottom": 20,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <View
      active={false}
      completed={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderColor": "#7981B2",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Text
        active={false}
        completed={true}
        style={
          {
            "color": "#7981B2",
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        1
      </Text>
    </View>
    <Text
      active={false}
      completed={true}
      style={
        {
          "color": "#7981B2",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Planes y coberturas
    </Text>
  </View>
  <Text
    completed={true}
    style={
      {
        "color": "#7981B2",
        "fontSize": 20,
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      }
    }
  >
    - - - -
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <View
      active={true}
      completed={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4F4FFF",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderColor": "#4F4FFF",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Text
        active={true}
        completed={false}
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        2
      </Text>
    </View>
    <Text
      active={true}
      completed={false}
      style={
        {
          "color": "#141938",
          "fontSize": 16,
          "fontWeight": "700",
        }
      }
    >
      Resumen
    </Text>
  </View>
</View>
`;

exports[`StepIndicator Component Snapshot tests matches snapshot for mobile view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderBottomColor": "#D7DBF5",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "gap": 10,
      "justifyContent": "flex-start",
      "paddingBottom": 20,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderColor": "#CBD5E1",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <Image
        source="arrorDownGrey"
      />
    </View>
  </View>
  <Text
    active={true}
    completed={false}
    style={
      {
        "color": "#141938",
        "fontSize": 10,
        "fontWeight": "700",
      }
    }
  >
    PASO 1 DE 2
  </Text>
  <View
    style={
      {
        "backgroundColor": "#C5CAE9",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 8,
        "overflow": "visible",
        "position": "relative",
        "width": "70%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#4F4FFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "5%",
        }
      }
    />
  </View>
</View>
`;
